app/services/tsheets_service.rb

```diff
@@ -0,0 +1,29 @@
+class TsheetsService
```

Comment:
I was in two minds about this, so appreciate the comments. Your last point sums up my reasoning, as far as the name goes, it's just named after the adapter pattern (https://en.wikipedia.org/wiki/Adapter_pattern). I think it's fairly standard (as far as their use goes) to keep adapters in `/app`.

I agree with the sentiment about too many unnecessary abstractions, and am not a fan of just using patterns 'because you can'.

You're right, it's a lossy abstraction, but we need to provide an interface somehow to tsheets that encapsulates the connection, whether it be in the adapter or in the service, so unless we come up with a method of exposing all of the functionality of the api in the adapter/service, then we would have to explicitly expose the functionality we need anyway.

The reasoning for is separation of concerns - the adapter deals with anything to do with the connection to the API, the service is for more complicated operations on tsheets objects.

The reasoning against is reduced complexity. 

At the moment my adapter implementation is very naive (and probably not very good ;)), so we probably wouldn't lose a lot by merging with the service. My gut tells me this is a bad move because we'd be violating the single responsibility principle, but as I said before I've been wavering on this myself.

That's my thoughts so far anyway, so real conclusion as yet. ;-)